opm/simulators/wells/MultisegmentWellEval.cpp

```diff
@@ -217,22 +217,27 @@ assembleDefaultPressureEq(const int seg,
                           WellState& well_state)
 {
     assert(seg != 0); // not top segment
+    const int seg_upwind = segments_.upwinding_segment(seg);
+    const bool reverseFlow = seg != seg_upwind;
 
     // for top segment, the well control equation will be used.
     EvalWell pressure_equation = primary_variables_.getSegmentPressure(seg);
+    EvalWell extra_derivatives;
 
     // we need to handle the pressure difference between the two segments
     // we only consider the hydrostatic pressure loss first
     // TODO: we might be able to add member variables to store these values, then we update well state
     // after converged
-    const auto hydro_pressure_drop = segments_.getHydroPressureLoss(seg);
+
     auto& ws = well_state.well(baseif_.indexOfWell());
     auto& segments = ws.segments;
-    segments.pressure_drop_hydrostatic[seg] = hydro_pressure_drop.value();
-    pressure_equation -= hydro_pressure_drop;
+    //pressure_equation -= hydro_pressure_drop;
 
     if (this->frictionalPressureLossConsidered()) {
-        const auto friction_pressure_drop = segments_.getFrictionPressureLoss(seg);
+        const auto friction_pressure_drop = segments_.getFrictionPressureLoss(seg, false);
+        if (reverseFlow){
+            extra_derivatives = -segments_.getFrictionPressureLoss(seg, true);
```

Comment:
The following code should go into this bracket? or at least inside the bracket (`if (this->frictionalPressureLossConsidered()) {`)
```c++
        MultisegmentWellAssemble<FluidSystem,Indices,Scalar>(baseif_).
            assemblePressureEqExtraDerivatives(seg, seg_upwind, extra_derivatives, linSys_);
```

Reply:
In the current form, if there is no frictional pressure loss, calling the function `assemblePressureEqExtraDerivatives` can be dangerous?

Reply:
Yes, moved `assemblePressureEqExtraDerivatives` up into the `(if (this->frictionalPressureLossConsidered())  `-bracket